peps/pep-0759.rst

```diff
@@ -0,0 +1,472 @@
+security issues in externally hosted wheels, and the same remediation process should be
+used.  In addition, since organizations with external hosting enabled MUST provide a
+support contact URI, that URI can be used in some cases to report the security issue to
+the hosting organization.  Such organization reporting won't make sense for malware, but
+could indeed be a very useful way to report security vulnerabilities in externally hosted
+wheels.
+
+Rejected ideas
+==============
+
+Several ideas were considered and rejected.
+
+- Requiring digital signatures on externally hosted wheel files, either in
+  addition to or other than hashes. We deem this unnecessary since the
+  checksum requirement should be enough to validate that the metadata on PyPI
+  for a wheel exactly matches the downloaded wheel.  The added complexity of
+  key management outweighs any additional benefit such digital signatures
+  might convey.
+- Hash verification on ``.rim`` file uploads. PyPI *could* verify that the
+  hash in the uploaded ``.rim`` file matches the externally hosted wheel
+  before it accepts the upload, but this requires downloading the external
+  wheel and performing the checksum. This would impose an ordering restriction
+  on uploads, such that the external wheel file is uploaded and publicly
+  available *before* uploading the ``.rim`` file to PyPI. While this might be
+  a good idea to avoid any race condition, it should not be a
+  requirement. This also means that the upload of the ``.rim`` file cannot be
+  accepted until this external ``.whl`` file is downloaded and verified. This
+  increases PyPI bandwidth and slows down the upload query, although
+  :pep:`694` draft uploads could potentially mitigate these concerns. Still,
+  the benefit is not likely worth the additional complexity.
+- Periodic verification of the download URLs by the index. PyPI could try to periodically
+  ensure that the external wheel host or the external ``.whl`` file itself is still
+  available, e.g. via an ``HTTP HEAD`` request. This is likely overkill and without also
+  providing the file's checksum in the response [#fn2]_, may not provide much additional
+  benefit.
+- This PEP could allow for an organization to provide fallback download hosts,
+  such that a secondary is available if the primary goes down.  We believe
+  that DNS-based replication is a much better, well-known technique, and
+  probably much more resilient anyway.
+- ``.rim`` file replacement. While it is allowed for ``.whl`` files to replace
+  existing ``.rim`` files, as long as a) the ``.rim`` file hasn't been deleted
+  or yanked, b) the checksums match, we do not allow replacing ``.whl`` files
+  with ``.rim`` files, nor do we allow a ``.rim`` file to overwrite an
+  existing ``.rim`` file. This latter could be a technique to change the
+  hosting URL for an externally hosted ``.whl``, however we do not think this
```

Comment:
```suggestion
  hosting URL for an externally hosted ``.whl``, but we do not think this
```

Reply:
Both [but and however](https://justpublishingadvice.com/but-and-however/) work, but in this case I prefer:

> While but allows your sentences to flow without interruption, however can give the sense of introducing a slight break, giving the reader a moment to absorb the shift in thought.

I appreciate the suggestion, however I think I'll reject it.  😄

Reply:
It's fine to use "however", but the grammar of the sentence doesn't work here. Instead use:
```suggestion
  hosting URL for an externally hosted ``.whl``; however, we do not think this
```